api/src/main/java/com/anysoftkeyboard/api/KeyCodes.java

```diff
@@ -66,5 +66,7 @@
     public static final int PRE_PREPARED_TEXT_POPUP = -151;
     public static final int PRE_PREPARED_EMAILS_POPUP = -152;
 
+    public static final int DELETE_RECENT_USED_SMILEYS = -160;
```

Comment:
Why do we need this to be a key-code? I don't see how we would like this functionality everywhere, right?

I imagine this to be similar to how we react to the `Settings` icon: https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/master/ime/app/src/main/java/com/anysoftkeyboard/quicktextkeys/ui/FrameKeyboardViewClickListener.java#L32 - have the key-press directly call `clearHistory()`

Reply:
I agree, but settings also seemed to be a key-code. https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/b68849e7a8f9714c1ff9d91acf49b7385f4d0d7b/api/src/main/java/com/anysoftkeyboard/api/KeyCodes.java#L39
This is why i added it to the list of key-codes.

Reply:
that's a different settings. The settings icon in the emoji layout navigates directly to the quick-text (emoji) settings. It does not use the `onKey` mechanism.

I suggest calling `clearHistroy()` directly there instead of calling `onKey` with the new key-code

Reply:
Okay, il will fix this tonigh.

Reply:
It seems not to be working when not using ```onKey```. 

It seems be linked with the ```getQuickKeyHistoryRecords()``` that i am able to fetch in the ```AnySoftKeyboard.java``` class, but not in ```FrameKeyboardViewClickListener.java```

I tried to use ```new QuickKeyHistoryRecords(AnyApplication.prefs(v.getContext())).clearHistory();``` instead of onKey, but it's not deleting anything. Weird